tests/testutil/service.go

```diff
@@ -202,6 +203,98 @@ func (ts *TestService) GetRunningPods() string {
 	return out
 }
 
+// GetEndpointURL returns the URL for this service's app if the App object is versioned it returns
+// the versioned url.
+func (ts *TestService) GetEndpointURL() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	url, err := ts.waitForEndpointDNS()
+	if err != nil {
+		ts.T.Fatalf("Failed to get the endpoint url:  %v", err.Error())
+		return ""
+	}
+	return url
+}
+
+// GetKubeEndpointUrlByRevision returns the versioned endpoint URL by using the revision provided
+// and returns it as string
+func (ts *TestService) GetKubeEndpointURL() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	args := []string{"--format", "json",
+		fmt.Sprintf("%s:%s",
+			strings.Split(ts.Name, ":")[0],
+			strings.Split(ts.Name, ":")[1])}
```

Comment:
Confused at what this Sprintf does, it splits a string on colon and then re-adds it back with colon ?

Reply:
It made more sense back when the method had the revision parameter.